daemonset/kubeadm-kuberouter-all-features-hostport.yaml

```diff
@@ -0,0 +1,184 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-router-cfg
+  namespace: kube-system
+  labels:
+    tier: node
+    k8s-app: kube-router
+data:
+  cni-conf.json: |
+    {
+       "cniVersion":"0.3.0",
+       "name":"mynet",
+       "plugins":[
+          {
+             "name":"kubernetes",
+             "type":"bridge",
+             "bridge":"kube-bridge",
+             "isDefaultGateway":true,
+             "ipam":{
+                "type":"host-local"
+             }
+          },
+          {
+             "type":"portmap",
+             "capabilities":{
+                "snat":true,
```

Comment:
is `snat` a requirement to use this?

Reply:
I think so. Though its not clear from https://github.com/containernetworking/plugins/tree/master/plugins/meta/portmap

Reply:
Seems like it sets up SNAT chain, might be good to double check if it conflicts with any SNAT chains kube-router creates

Reply:
@andrewsykim ok I checked, there is no conflict with any other rules.

```
Chain PREROUTING (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination
    7   468 CNI-HOSTPORT-DNAT  all  --  *      *       0.0.0.0/0            0.0.0.0/0            ADDRTYPE match dst-type LOCAL

Chain POSTROUTING (policy ACCEPT 24 packets, 1440 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 CNI-HOSTPORT-SNAT  all  --  *      *       127.0.0.1           !127.0.0.1

Chain CNI-DN-1d1e1acf7cbbe4ef6121a (1 references)
 pkts bytes target     prot opt in     out     source               destination
    1    60 DNAT       tcp  --  *      *       0.0.0.0/0            0.0.0.0/0            tcp dpt:8086 to:10.1.1.32:80

Chain CNI-HOSTPORT-DNAT (2 references)
 pkts bytes target     prot opt in     out     source               destination
  135  8148 CNI-DN-1d1e1acf7cbbe4ef6121a  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* dnat name: "mynet" id: "2fd850ee20c7105656a5b5ccd4669c04f5a9ed8035f2f35c41a5aa7e6cf4e81f" */

Chain CNI-HOSTPORT-SNAT (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 CNI-SN-1d1e1acf7cbbe4ef6121a  all  --  *      *       0.0.0.0/0            0.0.0.0/0            /* snat name: "mynet" id: "2fd850ee20c7105656a5b5ccd4669c04f5a9ed8035f2f35c41a5aa7e6cf4e81f" */

Chain CNI-SN-1d1e1acf7cbbe4ef6121a (1 references)
 pkts bytes target     prot opt in     out     source               destination
    0     0 MASQUERADE  tcp  --  *      *       127.0.0.1            10.1.1.32            tcp dpt:80
```